sr_logging_common/logging/check_log_size.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Shadow Robot Company Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import shutil
+import rospy
+import logging_utils as utils
+
+
+if __name__ == '__main__':
+    rospy.init_node('check_log_size')
+
+    additional_required_disk_space = rospy.get_param('~additional_required_disk_space', 10) * utils.GIGABYTE
+    allocated_log_space = rospy.get_param('~allocated_log_space', 10) * utils.GIGABYTE
+    log_growth_headroom = rospy.get_param('~log_growth_headroom', 5) * utils.GIGABYTE
+
+    total_disk_space, _, free_disk_space = shutil.disk_usage(utils.LOG_PATH)
+    logs_size = utils.get_directory_size(utils.LOG_PATH)
+
+    if free_disk_space < log_growth_headroom + max(logs_size, allocated_log_space) + additional_required_disk_space:
+        required_disk_space = log_growth_headroom + max(logs_size, allocated_log_space) + additional_required_disk_space
+        rospy.logfatal("Not enough free disk space to safely store roslogs:\n" +
+                       f"\tFree Disk Space: {free_disk_space / utils.GIGABYTE:.3f} of"
+                       f" {total_disk_space / utils.GIGABYTE:.3f} GB free\n" +
+                       f"\tRequired Free Disk Space: {required_disk_space / utils.GIGABYTE:.3f} GB\n\t\tMinimum" +
+                       f" Required Free Disk Space: {additional_required_disk_space / utils.GIGABYTE:.3f} GB\n" +
+                       f"\t\tAllocated Log Space: {allocated_log_space  / utils.GIGABYTE:.3f} GB\n" +
+                       f"\t\tLog Growth Headroom: {log_growth_headroom  / utils.GIGABYTE:.3f} GB\n" +
+                       f"\t\tCurrent Log Size: {logs_size / utils.GIGABYTE:.3f} GB")
+
+        if logs_size > allocated_log_space:
+            rospy.logwarn("Current size of logs exceeds allocated log space. Consider running clean_logs.py in" +
```

Comment:
Maybe print out the actual command they can use so that they can copy and paste it?

Reply:
https://github.com/shadow-robot/common_resources/pull/265/commits/f505c1ce6d6b2edf4ba9ff3dc6547c0049c7dbda